habitat-lab/habitat/config/CONFIG.md

```diff
@@ -0,0 +1,482 @@
+        type: DistanceToGoal
+        distance_to: POINT
+      success:
+        type: Success
+        success_distance: 0.2
+      spl:
+        type: SPL
+      distance_to_goal_reward:
+        type: DistanceToGoalReward
+    goal_sensor_uuid: pointgoal_with_gps_compass
+    count_obj_collisions: true
+    settle_steps: 5
+    constraint_violation_ends_episode: true
+    constraint_violation_drops_object: false
+    force_regenerate: false
+    should_save_to_cache: true
+    must_look_at_targ: true
+    object_in_hand_sample_prob: 0.167
+    render_target: true
+    ee_sample_factor: 0.2
+    ee_exclude_region: 0.0
+    base_angle_noise: 0.15
+    base_noise: 0.05
+    spawn_region_scale: 0.2
+    joint_max_impulse: -1.0
+    desired_resting_position:
+    - 0.5
+    - 0.0
+    - 1.0
+    use_marker_t: true
+    cache_robot_init: false
+    success_state: 0.0
+    easy_init: false
+    should_enforce_target_within_reach: false
+    task_spec_base_path: habitat/task/rearrange/pddl/
+    task_spec: ''
+    pddl_domain_def: replica_cad
+    obj_succ_thresh: 0.3
+    art_succ_thresh: 0.15
+    robot_at_thresh: 2.0
+    filter_nav_to_tasks: []
+    actions:
+      stop:
+        type: StopAction
+      move_forward:
+        type: MoveForwardAction
+      turn_left:
+        type: TurnLeftAction
+      turn_right:
+        type: TurnRightAction
+  dataset:
+    type: PointNav-v1
+    split: train
+    scenes_dir: data/scene_datasets
+    content_scenes:
+    - '*'
+    data_path: data/datasets/pointnav/gibson/v1/{split}/{split}.json.gz
+  gym:
+    auto_name: ''
+    obs_keys: null
+    action_keys: null
+    achieved_goal_keys: []
+    desired_goal_keys: []
+```
+</details>
+
+- The configuration keys are lowercase and live under the `habitat` namespace.
+- Embodied AI task specifications are separated from the benchmark configs (more general configs that include
+  task config as one of their sub-configs in the Defaults List). Task configs are grouped in the `habitat/task`
+  Config Group (located in [habitat-lab/config/habitat/task](habitat/task) directory) and benchmark configs
+  are placed in the [habitat-lab/config/benchmark](benchmark) directory.
+- `agent.sensors` is renamed to `agent.sim_sensors`,  `task.sensors` is renamed to `task.lab_sensors`.
+- Actions, agents, measurements and sensors configs are not directly attached to the simulator, task or agent
+  config nodes but are added to `task.actions`, `simulator.agents`, `task.measurements`, `task.lab_sensors`,
+  `simulator.agent_name.sim_sensors` sub-nodes of type `Dict[str, CorrespondingConfigNodeType]`
+  that can be updated dynamically. Consequently, Output Config contains only those config nodes that are listed in
+  `task.actions`, `simulator.agents`, `task.measurements`, `task.lab_sensors`, `simulator.agents.agent_name.sim_sensors`
+  (not all possible actions, agents, measurements and sensors config nodes).
+- `task.possible_actions` is removed.
+
+### New features
```

Comment:
That would be good to mention that functionality is the result of the migration:
```suggestion
### New functionality enabled
```